### PR TITLE
Expose operator_sdk versions used in CI

### PR DESCRIPTION
**Pull request by vkmc**
Currently we use two different versions of the operator_sdk in our CI. One is being used for building the operator bundles (v0.x) and the other is being used for deploying with operator bundles (v1.x).

This change makes this fact more explicit across the code and also enables the possibility of configuring which versions to use.

### CODE REVIEW

File: build/metadata.sh [hunk 1]
```diff
@@ -9,6 +9,7 @@ OPERATOR_NAME=${OPERATOR_NAME:-service-telemetry-operator}
 IMAGE_BUILDER=${IMAGE_BUILDER:-podman}
 IMAGE_BUILD_ARGS=${IMAGE_BUILD_ARGS:-''}
 IMAGE_TAG=${IMAGE_TAG:-latest}
+# NOTE(vkmc) Where is this env var being used?
```
Comment (elfiesmelfie) [inline build/metadata.sh:12]: This script is used in generate_bundles.sh

Comment (vkmc) [inline build/metadata.sh:12]: I mean L13

File: build/stf-run-ci/tasks/main.yml [hunk 2]
```diff
@@ -68,9 +68,10 @@
     base_dir: "{{ playbook_dir }}"
   when: base_dir | length == 0
 
-- name: Get new operator sdk
+- name: Get operator_sdk_v1 (deploy from bundles)
   when: __local_build_enabled | bool or __deploy_from_bundles_enabled | bool or __deploy_from_index_enabled | bool
-  ansible.builtin.command: "{{ base_dir }}/get_new_operator_sdk.sh {{ new_operator_sdk_version }}"
+  ansible.builtin.command: "{{ base_dir }}/get_operator_sdk.sh {{ operator_sdk_v1 }}"
+  creates: "{{ base_dir }}/working/operator-sdk-{{ operator_sdk_v1 }}"
```
Comment (elfiesmelfie) [inline build/stf-run-ci/tasks/main.yml:74]: We could download both versions here at the same time

File: build/get_operator_sdk.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	if [[ "${VERSION}" =~ "v0" ]]; then
+		# naming scheme for v0.x is operator-sdk-$VERSION-$ARCH-$OS e.g. operator-sdk-v0.19.4-x86_64-linux-gnu
+		curl -L ${OPERATOR_SDK_DL_URL}/operator-sdk-${VERSION}-x86_64-linux-gnu -o ${REL}/working/operator-sdk-${VERSION}
+    else
+		# naming scheme for v1.x is operator-sdk_$OS-$ARCH e.g. operator-sdk_linux_amd64
+		curl -L ${OPERATOR_SDK_DL_URL}/operator-sdk_${OS}_${ARCH} -o ${REL}/working/operator-sdk-${VERSION}
+	fi
+	chmod +x ${REL}/working/operator-sdk-${VERSION}
+	rm -f ${REL}/working/operator-sdk
+	ln -s operator-sdk-${VERSION} ${REL}/working/operator-sdk
```
Comment (elfiesmelfie) [inline build/get_operator_sdk.sh:22]: This becomes unnecessary if you are explicitly referencing operator-sdk-$version

**Review by elfiesmelfie — CHANGES_REQUESTED** [review]
Looks good in theory. I need to test this locally as well

File: build/stf-run-ci/tasks/main.yml (hunk 2 above)
Comment (elfiesmelfie) [inline build/stf-run-ci/tasks/main.yml:74]:
```suggestion
  ansible.builtin.command: "{{ base_dir }}/get_operator_sdk.sh {{ operator_sdk_v1 }}" creates="{{ base_dir }}/working/operator-sdk-{{ operator_sdk_v1 }}"
```

OR

```suggestion
  ansible.builtin.command:
    cmd: |
       "{{ base_dir }}/get_operator_sdk.sh {{ operator_sdk_v1 }}"
    creates: "{{ base_dir }}/working/operator-sdk-{{ operator_sdk_v1 }}"
```

File: build/metadata.sh (hunk 1 above)
Comment (elfiesmelfie) [reply build/metadata.sh:12]: Yup, metadata.sh is used in generate_bundlew

File: build/get_operator_sdk.sh (hunk 3 above)
Comment (elfiesmelfie) [reply build/get_operator_sdk.sh:22]: Line 22, that is

File: build/metadata.sh (hunk 1 above)
Comment (vkmc) [reply build/metadata.sh:12]: Yeah... a quick search though it seems that REQUIRED_OPERATOR_SDK_VERSION is not being used anywhere and should be safe to remove.

Comment (vkmc) [reply build/metadata.sh:12]: Yeah... a quick search though it seems that REQUIRED_OPERATOR_SDK_VERSION is not being used anywhere and should be safe to remove.

Comment (leifmadsen) [reply build/metadata.sh]: A few others as well which aren't used too much any more:

* build/push_container2ocp.sh
* deploy/remove_stf.sh
* deploy/quickstart.sh

File: build/run-ci.yaml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 ---
 # run STF CI setup in CRC (already provisioned)
 - hosts: localhost
-  gather_facts: no
+  gather_facts: yes
```
Comment (leifmadsen) [inline build/run-ci.yaml:4]: Is this required or was this left as part of testing?

**Review by leifmadsen — APPROVED** [review]
Looks sane enough to me :)

File: build/run-ci.yaml (hunk 4 above)
Comment (elfiesmelfie) [reply build/run-ci.yaml:4]: It's needed because we use ansible_env for a default somewhere.

It will be removed again soon, hopefully

**Pull request merged**